binding-hs/src/Client.hs

```diff
@@ -0,0 +1,192 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE TypeOperators         #-}
+
+module Client
+  ( envCreate
+  , envListAll
+  , envReset
+  , envStep
+  , envActionSpaceInfo
+  , envActionSpaceSample
+  , envActionSpaceContains
+  , envObservationSpaceInfo
+  , envMonitorStart
+  , envMonitorClose
+  , envClose
+  , upload
+  , shutdownServer
+  , EnvID (..)
+  , InstID (..)
+  , Environment (..)
+  , Observation (..)
+  , Step (..)
+  , Outcome (..)
+  , Info (..)
+  , Action (..)
+  , Monitor (..)
+  , Config (..)
+  ) where
+
+import           Data.Aeson          (FromJSON, Object (..), ToJSON)
+import qualified Data.HashMap.Strict as HM (HashMap (..))
+import           Data.Proxy          (Proxy (..))
+import qualified Data.Text           as T (Text)
```

Comment:
Normally you would also `import Data.Text (Text)` so that you don't have to namespace your types. [Recently I've been creating `module Foo.Prelude`s](https://github.com/openai/gym-http-api/pull/26/files#diff-824cb3049a334def036b9a687f25f009) which reexports any custom prelude as well as common imports for the project.